src/public/js/rs.js

```diff
@@ -45,13 +45,214 @@ var RS = (function() {
+
+      // Follow the B-M iteration rules to update tau and gamma:
+      DThreshold = (i + 1) / 2;
+      if (delta == 0 || Ds[i] > DThreshold || (Ds[i] == DThreshold && !Bs[i])) {
+        // RULE A
+        taus[i + 1] = UTIL.mergeTypedArrays(taus[i], new Uint8Array([0]));
+        gammas[i + 1] = UTIL.mergeTypedArrays(gammas[i], new Uint8Array([0]));
+      } else {
+        // RULE B
+        var invDelta = UTIL.multInv(delta);
+        taus[i + 1] = UTIL.polynomialScale(sigmas[i], invDelta);
+        gammas[i + 1] = UTIL.polynomialScale(omegas[i], invDelta);
+
+        // Update the D tracker
+        Ds[i + 1] = i + 1 - Ds[i];
+
+        // If the equality case triggered, turn the B flag off
+        if (Ds[i] == DThreshold) {
+          Bs[i + 1] = false;
+        }
+      }
+    }
+
+    return {
+      errorLocationPoly: sigmas[n - k],
+      errorMagnitudePoly: omegas[n - k]
+    };
+  }
+
+  // Performs a Chien search on an error location polynomial to find the set of
+  // error locations.
+  // Input errorLocationPoly is a big-endian array representing a polynomial,
+  // i.e. [1,2,3] -> x^2 + 2x + 3
+  // Output is a set of integers in GF(256), each of which is an error location, GEN^power
+  // -> i.e. taking log3 of an element in the set would give you the error position,
+  // the power at which the error occurred.
+  function chienSearch(errorLocationPoly) {
+    var xis = [];
+    for (var v = 0; v < 256; v++) {
+      var res = UTIL.polynomialEval(errorLocationPoly, v);
+      if (res == 0) {
+        xis.push(UTIL.multInv(v));
+      }
+    }
+    return xis;
+  }
+
+  // Performs Forney's Formula on a set of error locations and an error magnitude
+  // polynomial to find the set of error magnitudes.
+  // Input errorLocations is a set of integers in GF(256)
+  // Input errorMagnitudePoly is a big-endian array representing a polynomial,
+  // i.e. [1,2,3] -> x^2 + 2x + 3
+  // Output errorMagnitudes is a set of integers in GF(256)
+  function forneysFormula(errorLocations, errorMagnitudePoly) {
+    var errorMagnitudes = errorLocations.map(function(xi, i) {
+      var invXi = UTIL.multInv(xi);
+      var numerator = UTIL.polynomialEval(errorMagnitudePoly, invXi);
+
+      var denominator = 1;
+      var xj;
+      var productTerm;
+      for (var j = 0; j < errorLocations.length; j++) {
+        xj = errorLocations[j];
+        if (j != i) {
+          productTerm = (1 ^ UTIL.fieldMult(xj, invXi));
+          denominator = UTIL.fieldMult(denominator, productTerm);
+        }
+      }
+
+      return UTIL.fieldDiv(numerator, denominator);
+    });
+
+    return errorMagnitudes;
+  }
+
+  // Repairs a received codeword using computed error locations and error magnitudes.
+  // Input received is an array of length n
+  // Inputs errorLocations and errorMagnitudes are arrays of matching length
+  // Return codeword is an array of length n
+  function repair(received, errorLocations, errorMagnitudes) {
```

Comment:
It could also make sense for this function to not return anything and to in-fact just repair the received array in place. Simply a design choice. More performance and maybe slightly prettier to just manipulate received, but I am not sure if that might be against typical JS style. Like in other languages you can kind of inform a developer that your function might modify its input by asking for a pointer vs a const pointer vs pass by value, but it seems those semantics don't exist in JS.